emacs-everywhere.el

```diff
@@ -475,8 +482,30 @@ Please go to 'System Preferences > Security & Privacy > Privacy > Accessibility'
   (pcase emacs-everywhere--display-server
     (`(x11 . ,_) (emacs-everywhere--app-info-linux-x11))
     (`(wayland . KDE) (emacs-everywhere--app-info-linux-kde))
+    (`(wayland . sway) (emacs-everywhere--app-info-linux-sway))
     (_ (user-error "Unable to fetch app info with display server %S" emacs-everywhere--display-server))))
 
+
+(defun emacs-everywhere--app-info-linux-sway ()
+  "Return information on the current active window, on a Linux Sway session."
+  (let* ((json-string (emacs-everywhere--call "sh" "-c" "swaymsg -t get_tree | jq '.. | select(.type?) | select(.focused==true)'"))
+         (json-object (json-read-from-string json-string))
+         (window-id (number-to-string (cdr (assoc 'id json-object))))
+         (app-name (cdr (assoc 'app_id json-object)))
```

Comment:
While using this branch I noticed that not all applications have an `app_id` on my system, leading to a `Wrong type argument: stringp, nil
` in `*Messages*`. 

I think the ones running with Xwayland do not have one (Slack in my case). They have `window_properties['class']` set though, so I could get them to work with this change:

```suggestion
         (app-name (cdr (assoc 'app_id json-object)))
         (app-name (if app-name app-name (cdr (assoc 'class (cdr (assoc 'window_properties json-object))))))
```